test/mam_jid_mini_SUITE.erl

```diff
@@ -0,0 +1,32 @@
+-module(mam_jid_mini_SUITE).
+-include_lib("eunit/include/eunit.hrl").
+-compile([export_all]).
+
+-define(ae(Expected, Actual), ?assertEqual(Expected, Actual)).
+
+all() -> [test_encode_decode_functionality].
+
+test_encode_decode_functionality(_Config) ->
+    PossibleDomainNames = [<<"a">>, <<"b">>, <<"c">>, <<"d">>, <<"e">>, <<"f">>],
+    PossibleUserNames = [<<"">> | PossibleDomainNames],
+    PossibleResourceNames = [<<"just:some@random/text here">> | PossibleUserNames],
+    PossibleJIDs = [{U, D, R, catch jid:make(U, D, R)} || U <- PossibleUserNames,
+                                                          D <- PossibleDomainNames,
+                                                          R <- PossibleResourceNames],
+    FailedJIDS = [ct:log("~nJID decoding failed:~n"
```

Comment:
Why not just crush the test instead of logging?

Reply:
I think because there can be more than one case failing.
If we crush, we would see only one of them.